pagerduty/resource_pagerduty_event_rule.go

```diff
@@ -0,0 +1,145 @@
+package pagerduty
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func resourcePagerDutyEventRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePagerDutyEventRuleCreate,
+		Read:   resourcePagerDutyEventRuleRead,
+		Update: resourcePagerDutyEventRuleUpdate,
+		Delete: resourcePagerDutyEventRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"action_json": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"condition_json": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"advanced_condition_json": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"catch_all": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func buildEventRuleStruct(d *schema.ResourceData) *pagerduty.EventRule {
+	eventRule := &pagerduty.EventRule{
+		Actions:   expandString(d.Get("action_json").(string)),
+		Condition: expandString(d.Get("condition_json").(string)),
+	}
+
+	if attr, ok := d.GetOk("advanced_condition_json"); ok {
+		eventRule.AdvancedCondition = expandString(attr.(string))
+	}
+
+	if attr, ok := d.GetOk("catch_all"); ok {
+		eventRule.CatchAll = attr.(bool)
+	}
+
+	return eventRule
+}
+
+func expandString(v string) []interface{} {
+	var obj []interface{}
+	err := json.Unmarshal([]byte(v), &obj)
+
+	if err != nil {
+		log.Printf(string(err.Error()))
```

Comment:
What do you think about returning the error here if JSON unmarshal fails?

Reply:
For consistency, I've adopted the same strategy used in the [extension resource](https://github.com/terraform-providers/terraform-provider-pagerduty/blob/e9bf5a56146b00cb2fef5af92720b1156a5265d7/pagerduty/resource_pagerduty_extension.go#L196). Would that work?